controllers/playlists.controller.js

```diff
@@ -0,0 +1,114 @@
+const { Playlist } = require('../models/playlist.model');
+const { extend } = require('lodash');
+
+const createNewPlaylist = async (req, res) => {
+	try {
+		let newPlaylist = req.body;
+		newPlaylist = new Playlist(newPlaylist);
+		newPlaylist = await newPlaylist.save();
+		newPlaylist = await newPlaylist
+			.populate({ path: 'videoList.videoId', populate: { path: 'tutorId' } })
+			.execPopulate();
+
+		res.status(201).json({ response: newPlaylist, success: true });
+	} catch (error) {
+		res.status(500).json({
+			success: false,
+			message: 'Something went wrong',
+			errorMessage: error.message,
+		});
+	}
+};
+
+const getPlaylistFromDb = async (req, res, next, id) => {
+	try {
+		const playlist = await Playlist.findById(id);
+		req.playlist = playlist;
+		next();
+	} catch (error) {
+		res.status(404).json({
+			success: false,
+			message: 'Something went wrong',
+			errorMessage: error.message,
+		});
+	}
+};
+
+const updatePlaylist = async (req, res) => {
+	try {
+		let { playlist } = req;
+		const playlistUpdates = req.body;
+		playlist = extend(playlist, playlistUpdates);
+		playlist = await playlist.save();
+		playlist = await playlist
+			.populate({ path: 'videoList.videoId', populate: { path: 'tutorId' } })
+			.execPopulate();
+		res.status(200).json({ response: playlist, success: true });
+	} catch (error) {
+		res.status(500).json({
+			success: false,
+			message: 'Something went wrong',
+			errorMessage: error.message,
+		});
+	}
+};
+
+const deletePlaylist = async (req, res) => {
+	try {
+		let { playlist } = req;
+		if (playlist.isDefault) {
+			throw new Error('Cannot delete default playlist');
+		}
```

Comment:
visited your video app and you are not providing an option to delete any default playlist right?
these lines of code are redundant, you think of removing

Reply:
I just thought what if anyone does delete request from the browser, hence provided a check here as well.

Reply:
Yes, we usually put checks on both the frontend and the backend. 
The frontend checks provide fast verification to the user and is good UX. But frontend can always be manipulated and if someone accesses the backend, then there should be checks on the backend too.